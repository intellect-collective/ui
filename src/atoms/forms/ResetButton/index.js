import React from 'react';
import Button from '../Button';
import { getWrapped } from '../_decorators/_utils';

const ResetButton = (props) => (
    <Button { ...props } type="reset" />
);
ResetButton.displayName = 'ResetButton';
ResetButton.wrapped = Button;
ResetButton.root = getWrapped(Button);

export default ResetButton;
