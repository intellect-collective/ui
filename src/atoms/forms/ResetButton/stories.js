import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Resetbutton from './index';

storiesOf('Atoms / Forms / ResetButton', module)
    .add('default', () => (<Resetbutton name="test-reset-1" onClick={ action('onClick') }>Clear</Resetbutton>));
