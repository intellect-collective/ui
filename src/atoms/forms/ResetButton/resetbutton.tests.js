import React from 'react';
import ResetButton from '.';
import {
    basicClickableTests
} from '../../../../test/utils';

const validProps = {
    name: 'test',
    onClick: () => {}
};

const expectedProps = {
    name: 'test',
    type: 'reset',
    onClick: expect.any(Function)
};

describe('ResetButton', () => {
    basicClickableTests(ResetButton, 'button', validProps, expectedProps);
    // rerenderSuppressionTests(ResetButton, validProps);

    it('Ignores button type', () => {
        const wrapper = mount(<ResetButton name="test" type="button" />);
        expect(wrapper.find('button').prop('type')).toEqual('reset');
    });
});
