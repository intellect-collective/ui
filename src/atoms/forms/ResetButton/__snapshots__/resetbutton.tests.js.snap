// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetButton Should accept valid props 1`] = `
<ResetButton
  name="test"
  onClick={[Function]}
>
  <FormContext
    name="test"
    onClick={[Function]}
    type="reset"
  >
    <Clickable
      name="test"
      onClick={[Function]}
      type="reset"
    >
      <Button
        name="test"
        onClick={[Function]}
        type="reset"
        value=""
      >
        <button
          name="test"
          onClick={[Function]}
          type="reset"
          value=""
        />
      </Button>
    </Clickable>
  </FormContext>
</ResetButton>
`;
